planner/core/exhaust_physical_plans.go

```diff
@@ -677,15 +677,26 @@ func (p *LogicalJoin) constructInnerTableScanTask(
 		Desc:            desc,
 	}.Init(ds.ctx, ds.blockOffset)
 	ts.SetSchema(ds.schema.Clone())
+	if rowCount <= 0 {
+		rowCount = float64(1)
+	}
+	selectivity := float64(1)
+	countAfterAccess := rowCount
+	if len(ts.filterCondition) > 0 {
+		var err error
+		selectivity, _, err = ds.tableStats.HistColl.Selectivity(ds.ctx, ts.filterCondition)
+		if err != nil || selectivity <= 0 {
+			logutil.BgLogger().Debug("unexpected selectivity, use selection factor")
+			selectivity = selectionFactor
+		}
+		countAfterAccess = rowCount / selectivity
```

Comment:
Add a upper bound for `countAfterAccess` and a comment for why divide selectivty here?

Reply:
`RowCount:     math.Min(1.0, countAfterAccess),` has set an upper bound, and for IndexScan, `ScaleByExpectCnt` can guarantee an upper bound.